packages/@sanity/portable-text-editor/src/editor/plugins/createWithSchemaTypes.ts

```diff
@@ -59,6 +60,13 @@ export function createWithSchemaTypes({
         const key = span._key || keyGenerator()
         Transforms.setNodes(editor, {...span, _type: schemaTypes.span.name, _key: key}, {at: path})
       }
+
+      // catches cases when the children are missing keys but excludes it when the normalize is running the node as the editor object
+      if (node._key === undefined && !_.isEqual(node, editor)) {
```

Comment:
I think it would be better here to do:

```
if (node._key === undefined && (node.path.length === 1 || node.path.length === 2)) {
```

First, we don't need to check for editor, which is by using lodash `isEqual` not that performant.
Also, we only evaluate the nodes we are interested in having a new key (which is block (path length 1) and block child (path length 2).

Reply:
@skogsmaskin good point on the lodash, but for the second part: Just to double check, I had assumed that this warning would have appeared in children that could be "deeper" in the PTE - if that is the case then we are simply limiting this to the first two "levels" within the PTE right? 🤔

Reply:
For each editor there are only block nodes (path length 1) and child nodes (path length 2) that should have this exact normalization.

Reply:
I see, I had thought that the "path" in this case was related to the number of levels that the children it could have (so it could just go on forever theoretically).

Since that's not the case, I've changed it to your suggestion :)
Thanks for the clarification!